programs/stake/src/stake_state.rs

```diff
@@ -842,6 +817,83 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
     }
 }
 
+/// After calling `validate_delegate_amount()`, this struct contains calculated values that are used
+/// by the caller.
+struct ValidatedDelegateInfo {
+    stake_amount: u64,
+}
+
+/// Ensure the stake delegation amount is valid.  This checks that the account meets the minimum
+/// balance requirements to delegate stake.  If not, return an error.
+fn validate_delegate_amount(
```

Comment:
```suggestion
/// balance requirements of delegated stake.  If not, return an error.
fn validate_delegated_amount(
```
I know you're using "delegate" as a verb, but I found this confusing due to ambiguity with "delegate" the noun, which means something different.

Reply:
Works for me! I went back and forth on this myself, so, happy to go along with whatever version you prefer.